synapse/rest/client/v2_alpha/shared_rooms.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Half-Shot
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+
+from synapse.api.errors import AuthError
+from synapse.http.servlet import RestServlet
+
+from ._base import client_patterns
+
+logger = logging.getLogger(__name__)
+
+
+class UserSharedRoomsServlet(RestServlet):
+    """
+    GET /user/{user_id}/shared_rooms/{other_user_id} HTTP/1.1
+    """
+
+    PATTERNS = client_patterns(
+        "/user/(?P<user_id>[^/]*)/shared_rooms/(?P<other_user_id>[^/]*)",
```

Comment:
This is more a question for the spec side, but why offer users the ability to choose the user_id that's put here if they're only allowed to put their own?

If it's for consistency's sake, I believe @ara4n has voiced that this was originally done for application services, but with AS tokens being a thing, there's no need to continue this behaviour in the future.

Reply:
Yeah, it was for consistency. Let's also take it to the MSC because more :eyes:

Reply:
Lemme know when a link for an MSC exists then :P

Reply:
Ah, it does exist. Could you link it in the PR description please.